### PR TITLE
[Validator] Add more precise PHPDoc on constraints

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/Validator/Constraints/Callback.php [hunk 1]
```diff
@@ -27,8 +27,8 @@ class Callback extends Constraint
     public $callback;
 
     /**
-     * @param string|string[]|callable|array<string,mixed>|null $callback The callback definition
-     * @param string[]|null                                     $groups
+     * @param callable-string|string[]|callable|array<string,mixed>|null $callback The callback definition
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Validator/Constraints/Callback.php]: When a string is passed, it's interpreted as a method name. Which is not a callable-string.

Comment (alexandre-daubois) [reply src/Symfony/Component/Validator/Constraints/Callback.php]: Oh yes, got it. Removed from `Choice` as well.

**Review by nicolas-grekas — APPROVED** [review]
Why not (but please do go as far as adding `non-empty-string` to every single string arguments in the codebase... :) )

**Review by nicolas-grekas — APPROVED** [review]
> please do go as far as adding non-empty-string to every single string arguments in the codebase

Oops, I meant "do NOT" :laughing:

**Comment by alexandre-daubois** [comment]
No worries, that's what I understood :smile:

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @alexandre-daubois.

**Pull request merged**